lib/mutant/mutator/node/block.rb

```diff
@@ -31,6 +31,10 @@ def mutate_body
           mutate_body_receiver
         end
 
+        def unconditional_loop?
+          send.eql?(s(:send, nil, :loop))
+        end
```

Comment:
@mbj Something weird is going on here. Somehow this formulation breaks something in `NodePredicates` for reasons I do not yet understand. I have found a couple of different ways of expressing the same idea that do work though:

```ruby
def unconditional_loop?
  _, method_name = *send
  method_name.equal?(:loop)
end
```

or

```ruby
def unconditional_loop?
  children.first.eql?(s(:send, nil, :loop))
end
```

Seems like something about the `send` child definition under mutation testing is causing a problem. I am happy to use a workaround such as above but it seems like there is a subtle problem somewhere and could cause other issues if it's not clearly understood so I'm leaving my original (failing) implementation here. LMK if you have any ideas. No rush though, I may not have time to come back to this for a couple of days.

Reply:
@mbj Did you have any insights on this failure / seen anything like it before? I can try to look into it more or just use a workaround but I was pretty surprised to the error only appearing in the `mutant` run on CI.

Reply:
@mbj Curious if you had any insights on this, when you get a chance.

Reply:
@mbj Strangely I can no longer reproduce this. Maybe something subtle got fixed since when I first opened this? Unfortunately I can't look back at the CI logs since they are gone now and I don't know how to reproduce the error. I think this is ready for review again.